pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test: TestRepeatableReadLogic_upsert_non_metamorphic failed

pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test.TestRepeatableReadLogic_upsert_non_metamorphic [failed](https://tanzanite.cluster.engflow.com/invocations/default/f8b15e54-db78-49a0-bc9b-215d9e4bea67?testReportRun=1&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbC1yZXBlYXRhYmxlLXJlYWQ6bG9jYWwtcmVwZWF0YWJsZS1yZWFkX3Rlc3Q=) on release-25.2 @ [4c66ca53ca72240e24224cd78cf4d89c3741b200](https://github.com/cockroachdb/cockroach/commits/4c66ca53ca72240e24224cd78cf4d89c3741b200):


```
=== RUN   TestRepeatableReadLogic_upsert_non_metamorphic
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRepeatableReadLogic_upsert_non_metamorphic2480432506
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[10:47:47] setting distsql_workmem='15798B';
[10:47:50] --- progress: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 5 statements
[10:47:53] --- progress: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 9 statements
[10:47:55] --- progress: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 10 statements
[10:52:32] --- progress: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 14 statements
[10:54:30] --- done: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic with config local-repeatable-read: 14 tests, 0 failures
    logic.go:4327: 
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic:70: error while processing
    logic.go:4327: 
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic:70: 
        expected success, but found
        (40003) operation "intent_resolver_ir_batcher.sendBatch" timed out after 1m0.008s (given timeout 1m0s): aborted in DistSender: result is ambiguous: context deadline exceeded
        dist_sender.go:2341: in deduceRetryEarlyExitError()
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestRepeatableReadLogic_upsert_non_metamorphic2480432506
--- FAIL: TestRepeatableReadLogic_upsert_non_metamorphic (413.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRepeatableReadLogic_upsert_non_metamorphic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51725

--- a/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic
+++ b/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic
@@ -1,5 +1,7 @@
 # LogicTest: !metamorphic-batch-sizes
 
+skip under race
+
 # Regression test for UPSERT batching logic not respecting footprint-based
 # limiting (#102472).
 statement ok

